src/write_batch.rs

```diff
@@ -224,6 +224,8 @@ impl WriteBatch {
     /// Removes the database entries in the range ["begin_key", "end_key"), i.e.,
     /// including "begin_key" and excluding "end_key". It is not an error if no
     /// keys exist in the range ["begin_key", "end_key").
+    ///
+    /// `TxnDB` doesn't support `delete_range`.
```

Comment:
what happens `TxnDB` actually used `delete_range`? the librocksdb emits an runtime error unconditionally?

Reply:
if possible, `WriteBatch<D = DB>` and `WriteBatch<TxnDB>` might be more friendly?

Reply:
RocksDB will return `InvalidArguments` if `delete_range` is used on TransactionDB.

`WriteBatch<D = DB>` seems a good solution.